packages/@react-aria/tag/src/useTag.ts

```diff
@@ -10,74 +10,80 @@
  * governing permissions and limitations under the License.
  */
 
-import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, ReactNode} from 'react';
-import {DOMProps, Removable} from '@react-types/shared';
+import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent} from 'react';
 import {filterDOMProps, mergeProps, useId} from '@react-aria/utils';
+import {GridState} from '@react-stately/grid';
 // @ts-ignore
 import intlMessages from '../intl/*.json';
+import {TagProps} from '@react-types/tag';
+import {useGridCell, useGridRow} from '@react-aria/grid';
 import {useMessageFormatter} from '@react-aria/i18n';
 
 
-export interface AriaTagProps extends Removable<ReactNode, void>, DOMProps {
-  children?: ReactNode,
-  isDisabled?: boolean,
-  validationState?: 'invalid' | 'valid',
-  isSelected?: boolean,
-  role?: string
-}
-
 export interface TagAria {
   tagProps: HTMLAttributes<HTMLElement>,
-  labelProps: HTMLAttributes<HTMLElement>,
+  tagRowProps: HTMLAttributes<HTMLElement>,
   clearButtonProps: ButtonHTMLAttributes<HTMLButtonElement>
 }
 
-export function useTag(props: AriaTagProps): TagAria {
+export function useTag(props: TagProps<any>, state: GridState<any, any>): TagAria {
+  let {isFocused} = props;
   const {
     isDisabled,
-    validationState,
     isRemovable,
-    isSelected,
     onRemove,
     children,
-    role
+    item,
+    tagRef,
+    tagRowRef
   } = props;
   const formatMessage = useMessageFormatter(intlMessages);
   const removeString = formatMessage('remove');
-  const tagId = useId();
+  const labelId = useId();
```

Comment:
`labelId` is only returned in the `aria-labelledby` for `clearButtonProps` and is not assigned as an `id` for the Tag's  label.

Reply:
Will address in followup PR